pycheribuild/projects/cross/freertos.py

```diff
@@ -47,17 +47,35 @@ class BuildFreeRTOS(CrossCompileAutotoolsProject):
     needs_sysroot = False  # We don't need a complete sysroot
     supported_architectures = [
         CompilationTargets.BAREMETAL_NEWLIB_RISCV64_PURECAP,
-        CompilationTargets.BAREMETAL_NEWLIB_RISCV64]
+        CompilationTargets.BAREMETAL_NEWLIB_RISCV64,
+        CompilationTargets.BAREMETAL_NEWLIB_RISCV32_PURECAP,
+        CompilationTargets.BAREMETAL_NEWLIB_RISCV32]
     default_install_dir = DefaultInstallDir.SYSROOT
 
     # FreeRTOS Demos to build
     supported_freertos_demos = [
         # Generic/simple (CHERI-)RISC-V Demo that runs main_blinky on simulators
         # and simple SoCs
-        "RISC-V-Generic"]
+        "RISC-V-Generic",
+        # CHERI-RISC-V sophisticated Demo with more advanced device drivers
+        # and currently only runs on FPGA-GFE, purecap
+        "RISC-V_Galois_P1"]
 
     # Map Demos and the FreeRTOS apps we support building/running for
-    supported_demo_apps = {"RISC-V-Generic": ["main_blinky"]}
+    supported_demo_apps = {"RISC-V-Generic": ["main_blinky"],
+                           "RISC-V_Galois_P1":
+                           ["main_blinky", "main_netboot",
+                               # The followings haven't been tested for purecap
+                               "main_testgen",
+                               "main_uart_malware",
+                               "main_rtc",
+                               "main_sd",
+                               "main_udp",
+                               "main_peekpoke",
+                               "main_tcp",
```

Comment:
These got tested ages ago when we were working on the smart ballot box (though I forget whether peekpoke was made to derive capabilities or not, that might just have been the separate smart ballot box application that embedded an extended copy of it).

Reply:
I don't remember peekpoke needs work for CHERI when I made it run on `RISC-V-Generic`